Patch migrated from Desktop, where it was listed as:

pango-03-solaris-cjk-font-table.diff

This fixes the following CR.

15431111 - SUNBT6617438 [vermillion] Latin font should not be used for ASCII when monospace

This patch comes from in-house and this is Solaris specific patch because 
the array depends which fonts we install.
Here is the copy from the description of the original CR.
---
Created By: takao.fujiwara@sun.com
The problem is that when pango receives ASCII 'a', Japanese fonts won't be
choosed because PangoScriptForLang array does not have PANGO_SCRIPT_LATIN in
"ja".
However Our Japanese fonts include some scripts besides Japanese.

My suggestion is to maintain pango-script-lang-table.h in Sun since the array
depends which fonts we install.
---

--- pango-1.38.0/pango/pango-script-lang-table.h.orig	2015-04-30 18:56:44.000000000 +0000
+++ pango-1.38.0/pango/pango-script-lang-table.h	2016-08-31 13:02:31.501883298 +0000
@@ -105,7 +105,7 @@
   { "is",     { PANGO_SCRIPT_LATIN/*70*/ } },
   { "it",     { PANGO_SCRIPT_LATIN/*72*/ } },
   { "iu",     { PANGO_SCRIPT_CANADIAN_ABORIGINAL/*161*/ } },
-  { "ja",     { PANGO_SCRIPT_HAN/*2134*/, PANGO_SCRIPT_KATAKANA/*88*/, PANGO_SCRIPT_HIRAGANA/*85*/ } },
+  { "ja",     { PANGO_SCRIPT_LATIN, PANGO_SCRIPT_GREEK, PANGO_SCRIPT_CYRILLIC, PANGO_SCRIPT_HAN/*2134*/, PANGO_SCRIPT_KATAKANA/*88*/, PANGO_SCRIPT_HIRAGANA/*85*/ } },
   { "jv",     { PANGO_SCRIPT_LATIN/*56*/ } },
   { "ka",     { PANGO_SCRIPT_GEORGIAN/*33*/ } },
   { "kaa",    { PANGO_SCRIPT_CYRILLIC/*78*/ } },
@@ -116,7 +116,7 @@
   { "kl",     { PANGO_SCRIPT_LATIN/*81*/ } },
   { "km",     { PANGO_SCRIPT_KHMER/*63*/ } },
   { "kn",     { PANGO_SCRIPT_KANNADA/*70*/ } },
-  { "ko",     { PANGO_SCRIPT_HANGUL/*2442*/ } },
+  { "ko",     { PANGO_SCRIPT_LATIN, PANGO_SCRIPT_HANGUL/*2442*/ } },
   { "kok",    { PANGO_SCRIPT_DEVANAGARI/*68*/ } },
   { "kr",     { PANGO_SCRIPT_LATIN/*56*/ } },
   { "ks",     { PANGO_SCRIPT_ARABIC/*33*/ } },
@@ -250,10 +250,10 @@
   { "yi",     { PANGO_SCRIPT_HEBREW/*27*/ } },
   { "yo",     { PANGO_SCRIPT_LATIN/*114*/ } },
   { "za",     { PANGO_SCRIPT_LATIN/*52*/ } },
-  { "zh-cn",  { PANGO_SCRIPT_HAN/*6763*/ } },
-  { "zh-hk",  { PANGO_SCRIPT_HAN/*1083*/ } },
-  { "zh-mo",  { PANGO_SCRIPT_HAN/*1083*/ } },
-  { "zh-sg",  { PANGO_SCRIPT_HAN/*6763*/ } },
-  { "zh-tw",  { PANGO_SCRIPT_HAN/*13063*/ } },
+  { "zh-cn",  { PANGO_SCRIPT_LATIN, PANGO_SCRIPT_GREEK, PANGO_SCRIPT_CYRILLIC, PANGO_SCRIPT_HAN/*6763*/, PANGO_SCRIPT_KATAKANA, PANGO_SCRIPT_HIRAGANA } },
+  { "zh-hk",  { PANGO_SCRIPT_LATIN, PANGO_SCRIPT_GREEK, PANGO_SCRIPT_CYRILLIC, PANGO_SCRIPT_HAN/*1083*/, PANGO_SCRIPT_KATAKANA, PANGO_SCRIPT_HIRAGANA } },
+  { "zh-mo",  { PANGO_SCRIPT_LATIN, PANGO_SCRIPT_GREEK, PANGO_SCRIPT_CYRILLIC, PANGO_SCRIPT_HAN/*1083*/, PANGO_SCRIPT_KATAKANA, PANGO_SCRIPT_HIRAGANA } },
+  { "zh-sg",  { PANGO_SCRIPT_LATIN, PANGO_SCRIPT_GREEK, PANGO_SCRIPT_CYRILLIC, PANGO_SCRIPT_HAN/*6763*/, PANGO_SCRIPT_KATAKANA, PANGO_SCRIPT_HIRAGANA } },
+  { "zh-tw",  { PANGO_SCRIPT_LATIN, PANGO_SCRIPT_GREEK, PANGO_SCRIPT_CYRILLIC, PANGO_SCRIPT_HAN/*13063*/, PANGO_SCRIPT_KATAKANA, PANGO_SCRIPT_HIRAGANA } },
   { "zu",     { PANGO_SCRIPT_LATIN/*52*/ } }
 };
